### PR TITLE
create a new SGP number with a new cost code and a project number

### PR DESCRIPTION
**Pull request by sabrine33**

### CODE REVIEW

File: src/components/workAllocation/WorkAllocation.tsx [hunk 1]
```diff
@@ -309,18 +316,51 @@ export default function WorkAllocation() {
...
+                      value={values.project}
+                    />
+                    <AddButton
+                      data-testid="addNewProject-btn"
+                      className={errors.project ?? 'place-self-end'}
+                      onClick={(event) => {
+                        event.preventDefault();
+                        event.stopPropagation();
+                        setAddNewProjectCodeCode(true);
+                      }}
```
Comment (seenanair) [inline src/components/workAllocation/WorkAllocation.tsx:360]: Cost code add button is slightly mis-aligned (vertically) in UI
Screenshot 2024-02-12 at 12 46 09

File: src/components/workAllocation/WorkAllocation.tsx [hunk 2]
```diff
@@ -309,18 +316,51 @@ export default function WorkAllocation() {
...
+                      send({
+                        type: 'ADD_NEWLY_CREATED_PROJECT',
+                        project: projectFactory.build({ name: addNewConfigOptionInputRef.current!.value })
+                      });
+                      setFieldValue('project', addNewConfigOptionInputRef.current!.value);
+                    }}
+                    onFinish={() => {
+                      setAddNewProjectCodeCode(false);
+                    }}
+                    configLabel="Add New Project"
```
Comment (seenanair) [inline src/components/workAllocation/WorkAllocation.tsx:339]: Would it be better to just mention "New Project" as label?

File: src/components/workAllocation/WorkAllocation.tsx [hunk 3]
```diff
@@ -341,15 +381,52 @@ export default function WorkAllocation() {
...
+                      send({
+                        type: 'ADD_NEWLY_CREATED_COST_CODE',
+                        costCode: costCodeFactory.build({ code: addNewConfigOptionInputRef.current!.value })
+                      });
+                      setFieldValue('costCode', addNewConfigOptionInputRef.current!.value);
+                    }}
+                    onFinish={() => {
+                      setAddNewCostCode(false);
+                    }}
+                    configLabel="Add New Cost Code"
```
Comment (seenanair) [inline src/components/workAllocation/WorkAllocation.tsx:404]: Same as above, may be only "New Cost code"

File: cypress/e2e/pages/workAllocation.cy.ts [hunk 4]
```diff
@@ -266,6 +266,91 @@ describe('Work Allocation', () => {
...
+        cy.findByTestId('costCode').type('S12345');
+        cy.findByRole('button', { name: /Save/i }).click();
+      });
+      it('keeps the add new config option form', () => {
+        cy.findByLabelText('Add New Cost Code').should('be.visible');
+        cy.findByRole('button', { name: /Save/i }).should('be.visible');
+      });
+    });
+  });
+
```
Comment (seenanair) [inline cypress/e2e/pages/workAllocation.cy.ts:353]: Do we have a unit test cases for the input part of workAllocation page?  If so, these new test cases can be moved over there.

File: tests/unit/components/workAllocation/addNewConfigOption.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+      act(() => {
+        fireEvent.change(screen.getByTestId('Config-X'), { target: { value: 'new option' } });
+        fireEvent.click(screen.getByText('Save'));
+      });
+      await waitFor(() => {
+        expect(onFinish).not.toHaveBeenCalled();
+      });
+    });
+  });
+});
```
Comment (seenanair) [inline tests/unit/components/workAllocation/addNewConfigOption.spec.tsx:186]: Good set of unit test cases for the AddNewConfigOption component 👍

File: cypress/e2e/pages/workAllocation.cy.ts (hunk 4 above)
Comment (sabrine33) [reply cypress/e2e/pages/workAllocation.cy.ts:353]: the unit test case for the input are provided within the addNewConfigOption testing file, this is to test the interaction of the input file with the workAllocation component

**Review by seenanair — APPROVED** [review]
Looking good 👍

**Pull request merged**